pymongo/__init__.py

```diff
@@ -97,3 +98,44 @@ def has_c() -> bool:
         return True
     except ImportError:
         return False
+
+
+def timeout(seconds: Optional[float]) -> ContextManager:
+    """Apply the given timeout for a block of operations.
+
+    Use :func:`~pymongo.timeout` in a with-statement::
+
+      with pymongo.timeout(5):
+          client.db.coll.insert_one({})
+          client.db.coll2.insert_one({})
+
+    When the with-statement is entered, a deadline is set for the entire
```

Comment:
In the [PEP](https://peps.python.org/pep-0343/) is called a "with statement" with no hyphen, the hyphen confused me.

Reply:
For me it's much clearer with the hyphen since with is already a common word. It adds some mental overhead to read "... with" and then "... with statement". I wouldn't mind using `'with' statement` instead though.

Reply:
Even the pep you linked to uses `with-statement` interchangeable so I'm not going to change this.

Reply:
Fair enough